GunShipAlpha/src/GameState.java

```diff
@@ -25,14 +33,23 @@ boolean isEnemyBlocked(Enemy enemy) {
         return false;
     }
 
-    void handleProjectileCollision() {
+    boolean handleProjectileCollisionAndMoveEnemies() {
+        handleEnemyMovement(); // Move enemies first
+        boolean playerHit = false;
         ArrayList<Projectile> projectilesToRemove = new ArrayList<>();
         for (Projectile projectile : projectiles) {
             if (doesProjectileHitAnyEnemy(projectile)) {
                 handleHitForProjectile(projectile, projectilesToRemove);
             }
+            if (projectile.isCollidingWithPlayer()) {
+                System.out.println("Projectile hit the player");
+                playerHit = true;
+                System.out.println(playerHit);
```

Comment:
leftover

Reply:
if you mean the println was leftover I fixed it